app/src/main/kotlin/com/flxrs/dankchat/chat/ChatAdapter.kt

```diff
@@ -80,6 +80,51 @@ class ChatAdapter(
     inner class ViewHolder(val binding: ChatItemBinding) : RecyclerView.ViewHolder(binding.root) {
         val scope = CoroutineScope(Dispatchers.Main.immediate)
         val coroutineHandler = CoroutineExceptionHandler { _, throwable -> binding.itemText.handleException(throwable) }
+
+        fun bind(item: ChatItem, isAlternateBackground: Boolean) {
+            binding.replyGroup.isVisible = false
+            binding.itemText.alpha = when (item.importance) {
+                ChatImportance.SYSTEM  -> .75f
+                ChatImportance.DELETED -> .5f
+                ChatImportance.REGULAR -> 1f
+            }
+
+            when (val message = item.message) {
+                is PrivMessage            -> {
+                    if (message.thread != null && !item.isInReplies) {
+                        binding.replyGroup.isVisible = true
+                        val formatted = buildString {
+                            append(binding.itemReply.context.getString(R.string.reply_to))
+                            append(" @${message.thread.name}: ")
+                            append(message.thread.message)
+                        }
+                        binding.itemReply.text = formatted
+                        binding.itemReply.setOnClickListener { onReplyClick(message.thread.rootId) }
+                    }
+
+                    binding.itemText.handlePrivMessage(message, this, item.isMentionTab)
+                }
+
+                // Handle other message types if needed
+
+                else -> {
+                    // Handle other message types if needed
+                }
+            }
+
+            // Set the background color for the entire itemLayout
+            val background = when {
```

Comment:
Setting the background at the end will override background colors that should be visible instead of the checkered color in some cases. For example notice messages for new subscriptions should be shown purple. So I'm not sure how this is an improvement.

Reply:
1. Resolving the issues